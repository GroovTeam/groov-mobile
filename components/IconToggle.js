import React, { useState } from 'react';
import { TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const IconToggle = ({ onActivate, onDeactivate, startStatus, onIcon, offIcon, color, size }) => {

  const [toggled, setToggled] = useState(startStatus);

  const toggle = async () => {
    if (toggled)
      await onDeactivate();
    else
      await onActivate();

    setToggled(!toggled);
  };

  return (
    <TouchableOpacity
      testID='button'
      style={{marginBottom: 5}}
      onPress={toggle}
    >
      <Ionicons
        name={toggled ? onIcon : offIcon}
        size={size ? size : 45}
        color={color}
      />
    </TouchableOpacity>
  );
};

export default IconToggle;