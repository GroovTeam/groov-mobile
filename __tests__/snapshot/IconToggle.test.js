import React from 'react';
import { cleanup, render } from '@testing-library/react-native';

import IconToggle from '../../components/IconToggle';

describe('<IconToggle /> -- Snapshot', () => {
  afterEach(cleanup);

  it('renders properly', () => {
    const tree = render(
      <IconToggle
        onActivate={() => {}}
        onDeactivate={() => {}}
        onIcon={'heart'}
        offIcon={'heart-outlined'}
        color={'red'}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});