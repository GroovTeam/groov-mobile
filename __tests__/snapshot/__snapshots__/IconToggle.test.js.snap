// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconToggle /> -- Snapshot renders properly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 5,
      "opacity": 1,
    }
  }
  testID="button"
>
  <Text />
</View>
`;
